python/cudf/cudf/core/column/column.py

```diff
@@ -629,8 +625,8 @@ def __setitem__(self, key: Any, value: Any):
         """
 
         # Normalize value to scalar/column
-        value_normalized: cudf.Scalar | ColumnBase = (
-            cudf.Scalar(value, dtype=self.dtype)
+        value_normalized: plc.Scalar | ColumnBase = (
+            cudf.Scalar(value, dtype=self.dtype).device_value
```

Comment:
We can't easily remove this `cudf.Scalar`?

Reply:
Yeah not comfortable doing that yet in this PR.

`cudf.Scalar` has some preprocessing logic (converting an arbitrary scalar to a "cudf compatible" scalar) that needs to be pulled out first and applied here (and in other places as well)

Reply:
Ok sounds good